mono/metadata/class-init.c

```diff
@@ -1554,17 +1562,25 @@ count_virtual_methods (MonoClass *klass)
 		mcount = mono_class_get_method_count (klass);
 		for (i = 0; i < mcount; ++i) {
 			flags = klass->methods [i]->flags;
-			if (flags & METHOD_ATTRIBUTE_VIRTUAL)
+			if ((flags & METHOD_ATTRIBUTE_VIRTUAL))
+			{
+				if (method_is_reabstracted(flags))
```

Comment:
nit: space before `(`.  also in other places where you call `method_is_reabstracted`